component/azstorage/block_blob.go

```diff
@@ -1629,3 +1631,67 @@ func (bb *BlockBlob) SetFilter(filter string) error {
 	bb.Config.filter = &blobfilter.BlobFilter{}
 	return bb.Config.filter.Configure(filter)
 }
+
+func (bb *BlockBlob) updateMetadata(name string, metadata map[string]*string) error {
+	if len(metadata) == 0 {
+		log.Info("BlockBlob::updateMetadata: No metadata to update for %s", name)
+		return nil
+	}
+
+	blobClient := bb.Container.NewBlockBlobClient(filepath.Join(bb.Config.prefixPath, name))
+
+	// Fetch existing properties
+	prop, err := blobClient.GetProperties(context.Background(), &blob.GetPropertiesOptions{
+		CPKInfo: bb.blobCPKOpt,
+	})
+	if err != nil {
+		log.Err("BlockBlob::updateMetadata: Failed to fetch properties for %s [%s]", name, err.Error())
+		return err
+	}
+
+	// Update metadata if needed
+	if prop.Metadata == nil {
+		prop.Metadata = make(map[string]*string)
+	}
+
+	update := false
+	for key, value := range metadata {
```

Comment:
Why we are doing so much of processing. We can get the metadata and set uid/gid related keys irrespective of whether they are there or not and send the metadata back.